### PR TITLE
recover a mass session within the embedder portal, if applicable

### PR DESCRIPTION
**Pull request by siosonel**
## Description

To test:
- pull `master` from the sjpp branch
- follow the instructions at the top of http://localhost:3000/embedded.html.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by karishma-gangwani — APPROVED** [review]
works for me. all tests pass.

**Comment by xzhou82** [comment]
click a session link opens TWO tabs. 2nd new tab crashes. refresh on crashed tab restores to localhost:8000 showing correctly saved session

not sure if a quirk of that python server. i'm fine to merge and deploy to prod and see

**Comment by siosonel** [comment]
> click a session link opens TWO tabs. 2nd new tab crashes. refresh on crashed tab restores to localhost:8000 showing correctly saved session
> 
> not sure if a quirk of that python server. i'm fine to merge and deploy to prod and see

@xzhou82 @karishma-gangwani I'm able to reproduce the issue by disabling my browser cache. It should be fixed now, please test again. 

The issue is related to CloudFlare caching, it doesn't take into account that the server response may vary depending on the URL that originally sent a request. So, the CORS policy for the `/genomes` response is reused, even though the `Access-Control-Allow-Origin` value should have changed between the requests to the embedder portal versus the proteinpaint.stjude.org site.

**Review by xzhou82 — APPROVED** [review]
thanks!!

File: server/src/app.js [hunk 1]
```diff
@@ -127,6 +127,7 @@ app.use(express.static(__dirname+'/public'))
 export const basepath = serverconfig.basepath || ''
 
 function setHeaders(res) {
+	res.header('Vary', 'Origin')
```
Comment (xzhou82) [inline server/src/app.js:130]: helpful to comment on the purpose of these

**Pull request merged**